test/RibbonThetaVault.ts

```diff
@@ -1172,42 +1195,75 @@ function behavesLikeRibbonOptionsVault(params: {
 
         await vault.connect(ownerSigner).rollToNextOption();
 
+        let bidMultiplier = 2;
+
         const auctionDetails = await bidForOToken(
           gnosisAuction,
           optionsPremiumPricer,
           assetContract,
           userSigner.address,
-          params.expectedMintAmount,
+          defaultOtokenAddress,
           params.premium,
-          "2"
+          tokenDecimals,
+          bidMultiplier.toString()
         );
 
-        const vaultOTokenBalanceBefore = (
-          await defaultOtoken.balanceOf(vault.address)
-        ).toString();
-        await gnosisAuction
+        assert.equal(
+          (await defaultOtoken.balanceOf(vault.address)).toString(),
+          "0"
+        );
+
+        const assetBalanceBeforeSettle = await assetContract.balanceOf(
+          vault.address
+        );
+
+        let res = await gnosisAuction
           .connect(userSigner)
           .settleAuction(auctionDetails[0]);
-        const vaultOTokenBalanceAfter = (
-          await defaultOtoken.balanceOf(vault.address)
-        ).toString();
 
         assert.isAbove(
-          parseInt(vaultOTokenBalanceAfter),
-          parseInt(vaultOTokenBalanceBefore)
+          parseInt((await defaultOtoken.balanceOf(vault.address)).toString()),
+          parseInt(
+            params.expectedMintAmount
+              .div(bidMultiplier)
+              .mul(params.premiumDiscount.sub(1))
+              .div(1000)
+              .toString()
+          )
```

Comment:
If I do a placeSellOrder for say 5 tokens, and I am offering 1.5x than the minimum price per token, I actually get more than 5. In this case since we have a floor of premium * 997 / 1000 and my bidder actually pays premium, he gets ~3% more tokens so it will not exactly be 5 tokens. hence we do tokenBalanceAfterSettle > (mintAmount / 2 ) * 996 / 1000